### ISSUE TITLE
stageEndTime should be int?

### ISSUE THREAD

**Issue by richagain** [SoptikHa2/crapi-csharp-wrapper#19]
Hi, I ran into an issue when using the API rwapper 0.7.2.

public class ClanWarInfo
    {
        ....

        public int stageEndTime { get; set; }
    }

In the class ClanWarInfo, the stageEdnTime is int but not int?. However, when I get the data for some clan, stageEndTime is null.

The exception is as below:

Newtonsoft.Json.JsonSerializationException: Error getting value from 'stageEndTime' on 'CRAPI.ClanWarInfo'. ---> System.InvalidOperationException: Nullable object must have a value.

Could you try to fix this issue? 

Many thanks.

BTW, is it possible to make me the contributor so that I can change source code and push? I am working on an app which heavily depends on your wrapper.

**Comment by SoptikHa2** [comment]
Thank you for report, I'll get to it in about 12 hours. If you can repair it faster (make sure to run code in the example case - file program.cs (I should make proper tests)), submit pull request and I'll approve it with my mobile.

Even if I added you as contributor, you won't be able to upload updated package to nuget.

**Comment by SoptikHa2** [comment]
And one more thing, if you set breakpoint right after calling wrapper request (even if it throws exception), you can see `wrapper.ServerResponse`, it contains api response (json).

**Comment by richagain** [comment]
Thanks for your quick respond.

Turned out it was a small issue when there is no clan war for a clan, the stageEndTime was null.

The original code was (in ClanWarInfo.cs):

```
public int stageEndTime
        {
            get
            {
                return _warEndTime ?? _collectionEndTime.**Value**;
            }
            set { }
        }
```

I updated the code to:

```
public int stageEndTime
        {
            get
            {
                return _warEndTime ?? _collectionEndTime**.GetValueOrDefault()**;
            }
            set { }
        }
```

I got the issue fix reference to this link: https://stackoverflow.com/questions/10555682/nullable-object-must-have-a-value

I am very new to github and never submit push request. I will try to submit my code from my end, if it doesnot work, please try to update the code base on what I found. 

Many thanks

**Comment by richagain** [comment]
change Value to GetValueOrDefault()

**Comment by SoptikHa2** [comment]
I didn't test it, but shouldn't be the type be `int?` instead of `int`? I don't know how it works right now, but maybe it should be nullable.

**Comment by richagain** [comment]
I am not very sure, but I changed the type from int to int?, the issue was still there. Where I changed it to .GetValueOrDefault(), it worked. 

To provide the default (null), or the value of the int column. Or you can specify the default with an overload. Either way, the nullable currently doesnt have a value, which is why you get that exception.

**Comment by SoptikHa2** [comment]
Maybe this could be done with stageEndTime type `int?` and the getter `return _warEndTime ?? _collectionEndTime.Value ?? null`

I'm not sure what the values are, so I don't know, this is just an idea. Do it to make it work, I don't really care how, or wait few hours and I'll make it myself.

**Comment by SoptikHa2** [comment]
Thank you for your code, I've used it and the issue should be fixed in 2064410. I've uploaded the code to nuget and it should be available to download in few minutes.

If you want, you can write your code. I believe it should work if you download this repository, add new branch, change files and then create pull request. I recommend using [GitHub Desktop](https://desktop.github.com/) application for windows, which offers simple GUI to work with repositories.
